### PR TITLE
python310Packages.openllm: init at 0.2.22

### LINKED ISSUE #238587: Package request: OpenLLM
**Project description**

https://github.com/bentoml/OpenLLM

An open platform for operating large language models (LLMs) in production. Fine-tune, serve, deploy, and monitor any LLMs with ease. 

**Metadata**

* homepage URL:https://github.com/bentoml/OpenLLM
* source URL:https://github.com/bentoml/OpenLLM
* license: asl20
* platforms: unix, linux, darwin, ...

### PR DESCRIPTION
**Pull request by natsukium**
## Description of changes

Operating LLMs in production
https://github.com/bentoml/OpenLLM

The following PRs are the blocker.
- #246582
  - tracking information https://nixpk.gs/pr-tracker.html?pr=246582
- #249114

closes #238587

cc @happysalada @dit7ya 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
Other than the 2 comments I put, this looks good to me.

**Comment by natsukium** [comment]
It has been split into three modules, `openllm-core`, `openllm-client`, and `python-openllm`, in v0.2.27.

**Comment by natsukium** [comment]
> Other than the 2 comments I put, this looks good to me.

Sorry for the late reply.
These comments seem to have been lost.

File: pkgs/development/python-modules/openllm/default.nix [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+  nativeBuildInputs = [
+    hatch-fancy-pypi-readme
+    hatch-vcs
+    hatchling
+    pythonRelaxDepsHook
+  ];
+
+  pythonRemoveDeps = [
+    "cuda-python"
```
[the remark this answers is not in the document]
Comment (natsukium) [reply pkgs/development/python-modules/openllm/default.nix:63]: If we include cuda-python, openllm will depend on cuda, which will disadvantage CI and caching.

[the remark this answers is not in the document]
Comment (happysalada) [reply pkgs/development/python-modules/openllm/default.nix:63]: Then i would just add a comment about this, just to make it clear to someone coming from outside.

[the remark this answers is not in the document]
Comment (natsukium) [reply pkgs/development/python-modules/openllm/default.nix:64]: done

**Pull request merged**